### PR TITLE
disable MacOS regression tests on Jenkins

### PR DESCRIPTION
**Pull request by zacharyburnett**
MacOS regression tests have numerical differences from the Linux ones; additionally, the Boyle Jenkins machine is currently unstable.

**Checklist for maintainers**
- [N/A] ~added entry in `CHANGES.rst` within the relevant release section~
- [N/A] ~updated or added relevant tests~
- [N/A] ~updated relevant documentation~
- [N/A] ~added relevant milestone~
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [N/A] ~Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)~

### CODE REVIEW

File: JenkinsfileRT [hunk 1]
```diff
@@ -108,11 +108,11 @@ bc1.build_cmds = [
     "pip freeze",
 ]
 bc1.build_cmds = PipInject(env.OVERRIDE_REQUIREMENTS) + bc1.build_cmds
```
Comment (hbushouse) [inline JenkinsfileRT:110]: By leaving the `bc1.build_cmds` uncommented, does that mean it's still going to do the build, but just not actually run the tests? Or is the build not executed until `bc1.test_cmds` is executed?

**Comment by hbushouse** [comment]
> @hbushouse should I also disable the MacOS tests on the `JenkinsfileRT_dev` build as well?

Yes. That one's almost more important to shutdown than the regular RT, because that's the one that gets used multiple times a day by folks trying to test their PR's.  Whereas the regular RT typically only gets run once or twice a day. But it still should be removed from both for now.

**Comment by hbushouse** [comment]
> > @hbushouse should I also disable the MacOS tests on the `JenkinsfileRT_dev` build as well?
> 
> Yes. That one's almost more important to shutdown than the regular RT, because that's the one that gets used multiple times a day by folks trying to test their PR's. Whereas the regular RT typically only gets run once or twice a day. But it still should be removed from both for now.

Whoops, sorry. My bad. JenkinsfileRT is used by both our regular nightly RT runs and the Developer's PR branch runs. JenkinsfileRT_dev is only used by the weekly devdeps test. So perhaps we could leave it in JenkinsfileRT_dev, so that we at least attempt to get a Mac run once a week. So please ignore my previous reply.

File: JenkinsfileRT (hunk 1 above)
Comment (zacharyburnett) [reply JenkinsfileRT]: It should only run the build commands on `utils.run()`, so yes in this current config it will build and not test

Comment (hbushouse) [reply JenkinsfileRT]: So isn't it a waste of time to create the build and then not run it?  Why not also comment out the build?

File: JenkinsfileRT_dev [hunk 2]
```diff
@@ -103,11 +103,6 @@ bc1.build_cmds = [
     "pip freeze",
 ]
 bc1.build_cmds = PipInject(env.OVERRIDE_REQUIREMENTS) + bc1.build_cmds
-bc1.test_cmds = [
```
Comment (hbushouse) [inline JenkinsfileRT_dev:106]: Wait, I thought we had decided to leave the Mac build and test active in the `dev` file, which gets run once a week by the devdeps test.

Comment (zacharyburnett) [reply JenkinsfileRT_dev:106]: I realized that this should be copied from the `bc0` anyway, so having it here is redundant; I'll check that it runs in the job.

**Comment by zacharyburnett** [comment]
https://github.com/spacetelescope/jwst/blob/33f102e778fab17f34607c6e2303b5cac88382d0/JenkinsfileRT#L81

Is this still necessary? It might cause issues with the downstream ops build if these versions are not pinned to releases

File: JenkinsfileRT (hunk 1 above)
Comment (zacharyburnett) [reply JenkinsfileRT]: https://github.com/spacetelescope/jwst/blob/33f102e778fab17f34607c6e2303b5cac88382d0/JenkinsfileRT#L97-L99

I forgot, this is what creates the `macos-stable_deps.txt`, so it is necessary to have here

**Comment by zacharyburnett** [comment]
regtests at https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST-Developers-Pull-Requests/detail/JWST-Developers-Pull-Requests/648/pipeline

**Comment by hbushouse** [comment]
> https://github.com/spacetelescope/jwst/blob/33f102e778fab17f34607c6e2303b5cac88382d0/JenkinsfileRT#L81
> 
> Is this still necessary? It might cause issues with the downstream ops build if these versions are not pinned to releases

I thought that when we do releases we always create a special requirements file that has absolutely *everything* pinned to a specific version. We just like to let a few of them (stcal, stdatamodels) float here, so that we're always using the latest code for those particular dependencies.

**Comment by zacharyburnett** [comment]
> > https://github.com/spacetelescope/jwst/blob/33f102e778fab17f34607c6e2303b5cac88382d0/JenkinsfileRT#L81
> > 
> > Is this still necessary? It might cause issues with the downstream ops build if these versions are not pinned to releases
> 
> I thought that when we do releases we always create a special requirements file that has absolutely _everything_ pinned to a specific version. We just like to let a few of them (stcal, stdatamodels) float here, so that we're always using the latest code for those particular dependencies.

We do, `requirements-sdp.txt`. Unfortunately, however, in the build pipeline `requirements-sdp.txt` and / or `requirements-dev.txt` are installed *before* the `build_cmds` commands, so keeping these here would override the SDP pins in the final build environment. The files that are pushed to Artifactory and then deployed to DMS are derived from the final build environment in Jenkins, so don't necessarily align with `requirements-sdp.txt`

File: JenkinsfileRT [hunk 3]
```diff
@@ -78,8 +78,8 @@ bc0.pip_reqs_files = ['requirements-sdp.txt']
 bc0.build_cmds = [
     "pip install certifi -U --force-reinstall",
     "pip install -e .[test,sdp] --no-cache-dir",
-    "pip install git+https://github.com/spacetelescope/stdatamodels.git@master git+https://github.com/spacetelescope/stcal.git@main",
     "pip install pytest-xdist",
+    "pip install -r requirements-sdp.txt",
```
Comment (zacharyburnett) [inline JenkinsfileRT:82]: Actually, I can try just installing it here; if this works, then we can also use the master commits of the other dependencies.

**Comment by zacharyburnett** [comment]
regtests are blocked by #7543

**Comment by hbushouse** [comment]
#7543 is merged now, so you can rebase and try the tests.

**Comment by zacharyburnett** [comment]
regtest running at https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST-Developers-Pull-Requests/detail/JWST-Developers-Pull-Requests/652/pipeline

**Comment by hbushouse** [comment]
@zacharyburnett @WilliamJamieson Is it OK for me to go ahead and merge this, so that all of our other tests we're trying to run (e.g. on codecov changes) will run faster and cleaner?

**Comment by zacharyburnett** [comment]
> @zacharyburnett @WilliamJamieson Is it OK for me to go ahead and merge this, so that all of our other tests we're trying to run (e.g. on codecov changes) will run faster and cleaner?

yes it looks like this is going through successfully in the regtests, so we can merge it and sort out any issues later

**Review by hbushouse — APPROVED** [review]

**Pull request merged**